examples/custom_entities.rs

```diff
@@ -27,35 +28,41 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
     reader.trim_text(true);
 
     let mut buf = Vec::new();
-    let mut custom_entities = HashMap::new();
+    let mut custom_entities: HashMap<Vec<u8>, String> = HashMap::new();
     let entity_re = Regex::new(r#"<!ENTITY\s+([^ \t\r\n]+)\s+"([^"]*)"\s*>"#)?;
 
     loop {
+        let lookup_custom_entity = |ent: &_| custom_entities.get(ent).map(|s| s.as_str());
+
         match reader.read_event_into(&mut buf) {
             Ok(Event::DocType(ref e)) => {
                 for cap in entity_re.captures_iter(&e) {
-                    custom_entities.insert(cap[1].to_vec(), cap[2].to_vec());
+                    custom_entities.insert(cap[1].to_vec(), String::from_utf8(cap[2].to_vec())?);
```

Comment:
It is better to use `reader.decoder().decode()` for decoding. This is an example, I think, it is better to show the right way to get strings
```suggestion

                    let value = reader.decoder().decode(&cap[2])?.into_owned();
                    custom_entities.insert(cap[1].to_vec(), value);
```